.github/workflows/stack.yaml

```diff
@@ -0,0 +1,110 @@
+            stack-yaml: stack-coveralls.yaml
+    env:
+      STACK_YAML: stack.yaml
+      STACK_ARGS: '--resolver ${{ matrix.resolver }}'
+      cache-version: v2 # bump up this version to invalidate currently stored cache
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Cache
+      id: cache
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.stack
+          .stack-work
+        key: ${{ runner.os }}-${{ matrix.resolver }}-${{ env.cache-version }}
+        restore-keys: |
+          ${{ runner.os }}-${{ matrix.resolver }}-${{ env.cache-version }}
+    # Executable files somehow become incompatible after restoring on MacOS from a
+    # previous build, so it needs to be cleaned up. This is very inconvenient and will
+    # need to be fixed.
+    - name: MacOS workaround for failure due to setup-exe-cache
+      if: steps.cache.outputs.cache-hit == 'true' && matrix.os == 'macos-latest'
+      run: |
+        rm -r ~/.stack/setup-exe-cache
+        rm -r .stack-work
+    - name: Windows Cache
+      id: cache-windows
+      uses: actions/cache@v2
+      if: matrix.os == 'windows-latest'
+      with:
+        path: |
+          ~/AppData/Local/Programs/stack
+        key: ${{ runner.os }}-${{ matrix.resolver }}-programs-${{ env.cache-version }}
+        restore-keys: |
+          ${{ runner.os }}-${{ matrix.resolver }}-programs-${{ env.cache-version }}
+
+    # For ubuntu we can use fast ppa installation that also reduce cache size drastically
+    - name: Ubuntu install GHC
+      if: matrix.ghc != '' && matrix.os == 'ubuntu-latest'
+      run: |
+        sudo add-apt-repository ppa:hvr/ghc -y
+        sudo apt-get update
+        sudo apt-get install ghc-${{ matrix.ghc }}
+    - name: Install Stack and GHC
+      run: |
+        set -ex
+        curl -sSL https://get.haskellstack.org/ | sh -s - -f
+        curl -sSL https://gist.githubusercontent.com/lehins/fd36a8cc8bf853173437b17f6b6426ad/raw/4702d0252731ad8b21317375e917124c590819ce/git-modtime.hs -o git-modtime.hs
+        [ -n "${{ matrix.ghc }}" ] && [ "${{ matrix.os }}" == "ubuntu-latest" ] && STACK_ARGS="$STACK_ARGS --system-ghc"
+        [ -n "${{ matrix.stack-yaml }}" ] && STACK_YAML=${{ matrix.stack-yaml }}
+        stack --resolver ${{ matrix.resolver }} runghc git-modtime.hs
+    - name: Tests
+      env:
+        COVERALLS_TOKEN: ${{ secrets.COVERALLS_TOKEN }}
+      run: |
+        set -ex
+        [ -n "${{ matrix.ghc }}" ] && [ "${{ matrix.os }}" == "ubuntu-latest" ] && STACK_ARGS="$STACK_ARGS --system-ghc"
+        [ -n "${{ matrix.stack-yaml }}" ] && STACK_YAML=${{ matrix.stack-yaml }}
+        if [ "${{ matrix.os }}.${{ matrix.resolver }}" == "ubuntu-latest.lts-14" ] && [ -n "${COVERALLS_TOKEN}" ]; then
+          stack test random:spec random:legacy-test --coverage --haddock --no-haddock-deps
+          stack hpc report --all
+          curl -L https://github.com/lehins/stack-hpc-coveralls/releases/download/v0.0.6.0/shc.tar.gz | tar xz shc
```

Comment:
BTW is there something specific which blocks your fork of `stack-hpc-coveralls` from being merged upstream?

Reply:
Lack of maintainership, I guess. Note that until yesterday my fork was exactly the same as upstream. All I did was made a static binary release 0.0.5.0 that could be downloaded and used in CI (upstream release 0.0.4.0 doesn't work with newer stack versions), which I used for over a year now. Yesterday I added Github Actions support and made another release 0.0.6.0. 

We'll see maybe @rubik is still interested in keeping https://github.com/rubik/stack-hpc-coveralls up to date and make releases. I submitted a PR with updates here: https://github.com/rubik/stack-hpc-coveralls/pull/26

Reply:
Yeah, I used yours `shc.tar.gz` as well. Good stuff, it would be nice to grab it directly from Hackage.

Reply:
> it would be nice to grab it directly from Hackage.

I don't know. Building it from source vs a quick download of a static executable does not sound appealing to me. Especially when it is only needed in only one specific place. 

One way or another it does the exact job I need it for, so I am not too keen into putting much work into a tool like that.